message_center_compassion/mappings/compassion_mapping.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Flückiger Nathan <nathan.fluckiger@hotmail.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import models, fields, api
+
+
+class CompassionMapping(models.Model):
+
+    _name = "compassion_mapping"
+
+    name = fields.Char()
+    # Models and fieldToJson link
+    fields_json_ids = fields.Many2many('field_to_json')
+
+    # Using method
+
+    @api.multi
+    def data_to_json(self, data):
```

Comment:
Move method to abstract model. Then each object that has to be converted will inherit from the abstract model.
```
def data_to_json(self, mapping_name=None):
    mapping = self.env['compassion.mapping'].search([('model_id.name', '=', self._name), ('name', '=', mapping_name)]) # if mapping_name is defined
```
example of usage then:
    child.data_to_json('my_custom_mapping')